### PR TITLE
Vinod/basicjs

### PR DESCRIPTION
**Pull request by vinod9589**
basic js topic

### CODE REVIEW

File: Closure.html [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+<body>
+
+    <div style="width: 100%;height: 100vh;" >
+        <button id="grey" >grey</button>
+        <button id="yellow" >yellow</button>
+
+    </div>
+
+<script>
+     function clickhendler(color){
```
Comment (tw-vineet) [inline Closure.html]: function's name should be in camel case like( clickHandler ).check function names and change them.

File: Promises.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+// })
+
+
+function fun (){
+    console.log("hello")
+}
+
+
+ const result=setTimeout(function(){
```
Comment (tw-vineet) [inline Promises.js]: No need to use setTimeOut for API call. Remove setTimeOut.

File: Api.html
Comment (tw-vineet) [inline Api.html:1]: remove html file

File: Callback.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
+// A callback is a function passed as an argument to another function
+// This technique allows a function to call another function
+// A callback function can run after another function has finished
+
+
```
Comment (tw-vineet) [inline Callback.js]: Variable name should be informative. check all variable names and change accordingly.

File: Array.js [hunk 4]
```diff
@@ -0,0 +1,6 @@
+// The length property returns the length (size) of an array:
+const result=["vinod","shrikanth","adity"]
```
Comment (tw-vineet) [inline Array.js]: There you are using names, so you can assign variable like this 
const namesArr=["vinod","shrikanth","adity"]

File: vinodtasks/Api.html [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Document</title>
+</head>
+<body>
+
+    <div>
+
+    </div>
+
```
Comment (Ritikthoughtwin) [inline vinodtasks/Api.html]: what is the use of this file

File: vinodtasks/Array.js [hunk 6]
```diff
@@ -0,0 +1,6 @@
+// The length property returns the length (size) of an array:
+const result=["vinod","shrikanth","adity"]
+console.log(result.length)
+console.log(result.toString())
+// console.log(result.at(2))
+console.log(result.at[1])
```
Comment (Ritikthoughtwin) [inline vinodtasks/Array.js]: Add more Array methods

File: vinodtasks/Callback.js [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+
+
+// callback problem many time debit no control the code
+
+// function buynow(){
+//     transcation()
+//     transcation()
+  
+// }
+
```
Comment (Ritikthoughtwin) [inline vinodtasks/Callback.js]: Uncomment all this code and maintain code structure

File: vinodtasks/Closure.html [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+     function clickhendler(color){
+
+        return function(){
+        document.body.style.backgroundColor=`${color}`
+        }
+     }
+
+     document.getElementById("grey").onclick=clickhendler("grey")
+     document.getElementById("yellow").onclick=clickhendler("yellow")
+
```
Comment (Ritikthoughtwin) [inline vinodtasks/Closure.html]: Use only JS file to solve tasks.

File: vinodtasks/Closure.js [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+// addtwoo(5)
+// const addtwoo=function(num){
+//     return num+2
+// }
+
+// console.log(addtwo(5))
+// function addtwo(num){
+//     return num*5
+// }
+
```
Comment (Ritikthoughtwin) [inline vinodtasks/Closure.js:52]: Uncomment this code and use proper syntax for all examples

File: vinodtasks/Hoisting.js [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+var fn=function(){
+    console.log(a)
+    var a=20
+
+}
+console.log(fn())
+
+
+
+
```
Comment (Ritikthoughtwin) [inline vinodtasks/Hoisting.js:66]: Rework this file to improve readability and manage code.

File: vinodtasks/Promises.js [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+// },1000)
+
+// })
+// result.then(function(user){
+
+//    console.log(user)
+
+//     return 
+    
+
```
Comment (Ritikthoughtwin) [inline vinodtasks/Promises.js]: Refers previous comments

File: Closure.html (hunk 1 above)
Comment (vinod9589) [reply Closure.html]: delete the html file

File: Api.html
Comment (vinod9589) [reply Api.html:1]: delete the html file

File: Array.js (hunk 4 above)
Comment (vinod9589) [reply Array.js]: done variable names

File: vinodtasks/Callback.js (hunk 7 above)
Comment (vinod9589) [reply vinodtasks/Callback.js]: add more method and maintain code structure

File: vinodtasks/Closure.js (hunk 9 above)
Comment (vinod9589) [reply vinodtasks/Closure.js:52]: uncomment the code proper examples

File: vinodtasks/Promises.js (hunk 11 above)
Comment (vinod9589) [reply vinodtasks/Promises.js]: done all previous comments

File: Promises.js (hunk 2 above)
Comment (vinod9589) [reply Promises.js]: okay

File: vinodtasks/Api.html (hunk 5 above)
Comment (vinod9589) [reply vinodtasks/Api.html]: delete the html file

File: vinodtasks/Closure.html (hunk 8 above)
Comment (vinod9589) [reply vinodtasks/Closure.html]: i understand

File: vinodtasks/Hoisting.js (hunk 10 above)
Comment (vinod9589) [reply vinodtasks/Hoisting.js:66]: i have done

File: Callback.js (hunk 3 above)
Comment (vinod9589) [reply Callback.js]: i have done

**Comment by Ritikthoughtwin** [comment]
@vinod9589 when will you add the string tasks file?

**Review by Ritikthoughtwin — APPROVED** [review]

**Pull request merged**